### PR TITLE
Patch for issue  #30117

### PR DESCRIPTION
**Pull request by MadsRC**
Fixed the service beacon as it had a few issues.

The commit messages are pretty self explanatory.

The code is tested on 2015.8.3 - CentOS 7

### CODE REVIEW

File: salt/beacons/service.py [hunk 1]
```diff
@@ -72,15 +80,25 @@ def beacon(config):
         ret_dict = {}
         ret_dict[service] = {'running': __salt__['service.status'](service)}
 
+        # If no options is given to the service, we fall back to the defaults
+        # assign a False value to oncleanshutdown and onchangeonly. Those
+        # key:values are then added to the service dictionary.
+        if config[service] is None:
+            defaults = {
+                    'oncleanshutdown':False,
```
Comment (cachedout) [inline salt/beacons/service.py]: Please add whitespace after the colon on this line and the next.

**Comment by cachedout** [comment]
Hi @MadsRC 

This logic looks good. There is a minor style issue to clean up and then we will get this merged. Thanks very much!

File: salt/beacons/service.py [hunk 2]
```diff
@@ -72,21 +80,28 @@ def beacon(config):
...
-                ret_dict[service]['shutdown'] = 'clean'
-
+            ret_dict[service]['uncleanshutdown'] = True if os.path.exists(filename) else False
+            
         if 'onchangeonly' in config[service] and config[service]['onchangeonly'] is True:
-            if service not in last_status:
-                last_status[service] = ''
+            if service not in LAST_STATUS:
+                LAST_STATUS[service] = ''
             if last_status[service] != ret_dict[service]:
```
Comment (cachedout) [inline salt/beacons/service.py]: Did you need to capitalize this `last_service` too?

Comment (MadsRC) [inline salt/beacons/service.py]: I absolutely did not, that's what I get for using the editor on GitHub.
Fixed it and ran it through pylint, seems good now.

**Comment by basepi** [comment]
Thanks @MadsRC !

**Comment by basepi** [comment]
Backport to 2015.8

**Pull request merged**